datadog_checks_base/tests/base/checks/test_agent_check.py

```diff
@@ -1043,106 +1043,211 @@ def test_load_configuration_models(dd_run_check, mocker):
         ),
         pytest.param(
-            {'send_distribution_sums_as_monotonic': False, 'exclude_labels': True},
-            [('send_distribution_counts_as_monotonic', True), ('include_labels', True)],
+            {
+                "send_distribution_sums_as_monotonic": False,
+                "exclude_labels": True,
+            },
+            [
+                ("send_distribution_counts_as_monotonic", True),
+                ("include_labels", True),
+            ],
             None,
-            id='different options no typos',
+            [],
+            id="different options no typos",
         ),
         pytest.param(
-            {'send_distribution_count_as_monotonic': True, 'exclude_label': True},
+            {
+                "send_distribution_count_as_monotonic": True,
+                "exclude_label": True,
+            },
             [
-                ('send_distribution_sums_as_monotonic', False),
-                ('send_distribution_counts_as_monotonic', True),
-                ('exclude_labels', False),
-                ('include_labels', True),
+                ("send_distribution_sums_as_monotonic", False),
+                ("send_distribution_counts_as_monotonic", True),
+                ("exclude_labels", False),
+                ("include_labels", True),
             ],
             [
                 (
-                    'Detected potential typo in configuration option in test/instance section: '
-                    '`send_distribution_count_as_monotonic`. Did you mean send_distribution_counts_as_monotonic?'
+                    "Detected potential typo in configuration option in test/instance section: "
+                    "`send_distribution_count_as_monotonic`. Did you mean send_distribution_counts_as_monotonic?"
                 ),
                 (
-                    'Detected potential typo in configuration option in test/instance section: `exclude_label`. '
-                    'Did you mean exclude_labels?'
+                    "Detected potential typo in configuration option in test/instance section: `exclude_label`. "
+                    "Did you mean exclude_labels?"
                 ),
             ],
-            id='different options typo',
+            [
+                "send_distribution_count_as_monotonic",
+                "exclude_label",
+            ],
+            id="different options typo",
+        ),
+        pytest.param(
+            {
+                "field": "value",
+                "schema": "my_schema",
+            },
+            BaseModelTest(field="my_field", schema_="the_schema"),
+            None,
+            [],
+            id="using an alias",
+        ),
+        pytest.param(
+            {
+                "field": "value",
+                "schema_": "my_schema",
+            },
+            BaseModelTest(field="my_field", schema_="the_schema"),
+            None,
+            [],
+            id="not using an alias",
         ),
```

Comment:
Can you also write a test case where there is a typo in an alias?

Reply:
Good idea, added!

Reply:
thanks!